vtex/faststore
# feat: Integrate with Intelligent Search new attributes field

## What's the purpose of this pull request?

Integrate with IS new attribute field.

Example of attribute inside product item:
```
{
  "id": "2"
  "name": "Type",
  "value": "Wireless",
  "visible": true
}
```

## How it works?

Now it's possible to access the new IS field "attributes" through the "additionalProperty".

## How to test it?

Checkout to this branch and run `yarn dev`.
I've added an attribute to the SKU of ID equal to `61539541` on the `storeframework` account catalog. 

To test it you could add some `console.log` in components, for example:
In `ProductDetailsSettings`: `console.log('Additional property: ', additionalProperty)`
Go to `/fantastic-concrete-keyboard-61539541/p`
<img width="1181" alt="Screenshot 2024-03-05 at 09 08 30" src="https://github.com/vtex/faststore/assets/19983991/f7002559-acba-4f24-94d3-6c38c280ae8a">

In `ProductGalleryPage`: `console.log('Attributes: ', products[0]?.node?.additionalProperty)`
Go to `/s?q=keyboard&fuzzy=0&operator=and&facets=fuzzy%2Coperator&sort=score_desc&page=0`
<img width="1187" alt="Screenshot 2024-03-05 at 09 08 03" src="https://github.com/vtex/faststore/assets/19983991/b65c0834-6afe-46c4-849b-c3d446693bd7">

Every product should have this field, it'll be empty if there is no attribute (`[]`).
<img width="1179" alt="Screenshot 2024-03-05 at 10 33 37" src="https://github.com/vtex/faststore/assets/19983991/30f24315-7d90-413a-8dc4-e5d3c4b18cc7">

### Starters Deploy Preview

There is an issue with codesandbox, that is preventing me from installing the new core test version
<img width="985" alt="Screenshot 2024-03-04 at 10 42 49" src="https://github.com/vtex/faststore/assets/19983991/ab4419eb-d0b5-4a83-81d5-dac1b74deea0">

## References

[Jira task](https://vtex-dev.atlassian.net/browse/SFS-485)
